File: app/src/main/java/com/eshopthis/finds/Activities/Clearance.java
package com.eshopthis.finds.Activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.os.Bundle;
import com.eshopthis.finds.Adapter.Adapter;
import com.eshopthis.finds.R;
import com.eshopthis.finds.db.Model;
import java.util.ArrayList;
import java.util.List;

public class Clearance extends AppCompatActivity {

    RecyclerView clubsKeyRingRV;
    private Adapter adapter;
    private RecyclerView.LayoutManager layoutManager;
    private List<Model> dataList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_clearance);

        clubsKeyRingRV = findViewById(R.id.id_Clearance);
        fetchData();
    }

    private void fetchData() {
        layoutManager = new GridLayoutManager(getApplicationContext(), 3);
        clubsKeyRingRV.setLayoutManager(layoutManager);
        dataList = new ArrayList<>();
        adapter = new Adapter(this, dataList);
        clubsKeyRingRV.setAdapter(adapter);
        prepareData();
    }

    private void prepareData() {
        // Simulate adding items, ensure you have these drawable resources
        dataList.clear();
        dataList.add(new Model("1", "Clearance Item 1", "9.99", "Description for Clearance Item 1", R.drawable.cplus));
        dataList.add(new Model("2", "Clearance Item 2", "11.99", "Description for Clearance Item 2", R.drawable.cplus));
        // Add more items as needed
        adapter.notifyDataSetChanged();
    }
}

File: app/src/main/java/com/eshopthis/finds/Activities/Clearance.kt
package com.eshopthis.finds.Activities

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.eshopthis.finds.Adapter.Adapter
import com.eshopthis.finds.R
import com.eshopthis.finds.data.Model

class Clearance : AppCompatActivity() {

    private lateinit var clearanceRV: RecyclerView
    private lateinit var adapter: Adapter
    private lateinit var layoutManager: RecyclerView.LayoutManager
    private val dataList = mutableListOf<Model>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_clearance)

        clearanceRV = findViewById(R.id.id_Clearance)
        fetchData()
    }

    private fun fetchData() {
        layoutManager = GridLayoutManager(applicationContext, 2)
        clearanceRV.layoutManager = layoutManager
        adapter = Adapter(this, dataList)
        clearanceRV.adapter = adapter
        prepareData()
    }

    private fun prepareData() {
        dataList.clear()
        dataList.add(Model("1", "Clearance Book 1", "5.99", "Description for Clearance Book 1", R.drawable.clearance))
        dataList.add(Model("2", "Clearance Book 2", "7.99", "Description for Clearance Book 2", R.drawable.java))
        adapter.notifyDataSetChanged()
    }
}